Fuel key generation

e0ne opened the pull request:
Generate Fuel UUID for registration from Fuel UI

dmi-try committed:
Remove old versions of cobbler manifests for rhel

Commit:
Merge pull request #489 from aglarendil/master

FF do develop branch

aglarendil committed:
[PRD-1499] Assign keystone_ec2_url value

aglarendil committed:
remove obsolete debug in firewall module

Commit:
Merge pull request #501 from Mirantis/prd1302

remove obsolete debug in firewall module

Commit:
Merge pull request #498 from Mirantis/prd1499

[PRD-1499] Assign keystone_ec2_url value

r0mik committed:
add rubygem json for rhel

r0mik committed:
add rubygem json for rhel

e0ne committed:
Generete Fuel UUID during master node installation

e0ne committed:
Generate Fuel UUID using 'uuidgen' tool

Commit:
fix deploy

mirantis-jenkins commented:
:-1: Test FAILed.
Refer to this link for build results: http://fuel-jenkins.mirantis.com/job/fuel-syntax-check/31/

plechenko commented:
retest this please

mirantis-jenkins commented:
:-1: Test FAILed.
Refer to this link for build results: http://fuel-jenkins.mirantis.com/job/fuel-syntax-check/32/

plechenko commented:
retest this please

mirantis-jenkins commented:
:+1: Test PASSed.
Refer to this link for build results: http://fuel-jenkins.mirantis.com/job/fuel-syntax-check/33/

warpc committed:
Double var declaration fix for NailyFacts deploy

xarses commented:
Do we need to call this "generate_fuel_key"? wouldn't "generate_uuid_key" be more suited if anyone where to re-use this?

e0ne commented:
May be, but also we could change algorithm of Fuel key generation

Commit:
add necessary patched file to ceph-deploy

added files/ceph-deploy
added file for /usr/bin/ceph-deploy
cleaned some trailing whitespace

Commit:
fix ceph-deploy file; fix keystone class

Commit:
Added support for disk labels and unmounted partitions

bogdando commented on deployment/puppet/nailgun/lib/facter/generate_fuel_key.rb:
```diff
@@ -0,0 +1,5 @@
+Facter.add("generate_fuel_key") do
```
would it generate new key every time it have been called as $::generate_fuel_key ?

e0ne replied:
Yes, it will. Do we need to store it anywhere in fuel?

adanin replied:
Yes. What about to create a subtask about it (UUID generation inside of Fuel lib) and assign it to someone from Fuel lib team?

Commit:
fix ceph-deploy source; fix deploy push config

Commit:
Determine list of OSD devices dynamically

xdel committed:
Merge pull request #507 from r0mik/rhos_json

Add  rubygem-json to RHOS repos

Commit:
expanded readme; fix race in deps.pp

extended README with cinder examples
removed some todo's from site.pp
hard coded file require

angdraug committed:
not reinstall ntp in site.pp so that cobbler's ntp.conf changes are not overwritten

angdraug committed:
typo in ceph README: rdb -> rbd

Commit:
fix nova_compute

Commit:
fix readme whitespace for the ...

mihgen commented:
-1. Every time we re-run puppet for any reason, you will get different uuid. This is simply unacceptable.

I suggest another approach - let's generate uuid during anaconda post-install section. Thus, we can be sure it is generated once.

xarses commented:
Can we stash these like with the new sshkeys?

mihgen commented:
Umm how do we do it with the new sshkeys?

Commit:
replace tinyproxy to apache

mihgen committed:
Merge remote branch 'origin/develop'

xdel committed:
those files should be licensed for sure

xdel committed:
Merge pull request #515 from xdel/licenses

those files should be licensed for sure

dmi-try committed:
Merge remote-tracking branch 'origin/master' into no-rhel-ks

Conflicts:
	deployment/puppet/cobbler/manifests/distro/rhel63-x86_64.pp
	deployment/puppet/cobbler/manifests/distro/rhel63_x86_64.pp
	deployment/puppet/cobbler/manifests/profile/rhel63-x86_64.pp
	deployment/puppet/cobbler/manifests/profile/rhel63_x86_64.pp
	deployment/puppet/cobbler/templates/kickstart/rhel.ks.erb

xdel committed:
Merge pull request #514 from Zipfer/apache_api_proxy

replace tinyproxy to apache

Commit:
add synax check

Commit:
Get rid of set -e, set +e, etc. in syntax check script

mirantis-jenkins commented:
:-1: Test FAILed.
Refer to this link for build results: http://fuel-jenkins.mirantis.com/job/fuel_syntax_check/6/

xarses commented:
They are being created and stored in astute. see https://github.com/Mirantis/astute/pull/62 and https://groups.google.com/a/mirantis.com/forum/#!topic/fuel-core-team/J1RkK3J6_0o.

Maybe we can extend the generator to be able to generate and store other things such as UUID. This would also be helpful for where other items are being used as "secrets" and should be regenerated, but only for each cluster.

Commit:
Fix Anaconda issues with leftover LVM metadata prior to installation

dmitryilyin commented:
Wait, what are we going to do here?

You have just created a fact that return new random uuid each time facter runs and write this fact into settings.yaml template.

This is bad idea because uuid will change on every puppet run on master node... ok, we usually don't run puppet again but still this method is absolutely incorrect.

The right way to do it would be to execute 'uuidgen > /etc/fuel-uuid' in anaconda post-install script and then write a fact to get value from this file:

Facter.add('fuel_uuid') do
  setcode do
    begin
      IO.read('/etc/fuel_uuid')
    rescue
      nil
    end
  end
end
~  
Next you can add this uuid into settings template

bogdando commented:
We can as well use dmidecode -s system-uuid

mihgen commented:
How uniq is it?

On Wed, Sep 4, 2013 at 4:25 PM, Bogdan Dobrelya notifications@github.comwrote:

> We can as well use dmidecode -s system-uuid
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/Mirantis/fuel/pull/510#issuecomment-23785015
> .

## 

Mike Scherbakov

xdel committed:
Merge pull request #522 from Mirantis/develop

Update master for #492

xdel committed:
remove call to deleted class

xdel committed:
Merge pull request #525 from xdel/untiny

remove call to deleted class

Commit:
Fixed typo in LVM metadata commands

Commit:
fix syntax mysql replicator

Commit:
fix syntax ceph deploy.pp

Commit:
Merge pull request #518 from Zipfer/fuel_syntax_check

fuel synax check

sirmax123 committed:
added heat module

sirmax123 committed:
added savanna module

sirmax123 committed:
added murano module

bogdando commented:
For virtual cases, at least for single KVM host case, it would be, because it represents the virsh domain name which is always uniq.
For HW and other virtual cases, need to be investigated more precisely.

Here is ruby example for dmi system uuid extraction:
require 'rubygems'
require 'ohai'
@os = Ohai::System.new()
@os.all_plugins
uuid = @os[:dmi][:system][:uuid].strip rescue nil

mihgen commented:
We don't need UUID of each node. We need only one UUID for master node
installation.

On Thu, Sep 5, 2013 at 3:32 PM, Bogdan Dobrelya notifications@github.comwrote:

> For virtual cases, at least for single KVM host case, it would be, because
> it represents the virsh domain name which is always uniq.
> For HW and other virtual cases, need to be investigated more precisely.
> 
> Here is the example of custom fact for uuid would have been extracted from
> DMI on bootstrap stage:
> https://github.com/bogdando/fuelweb/blob/poc_fence/bin/agent#L281-L284
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/Mirantis/fuel/pull/510#issuecomment-23861279
> .

## 

Mike Scherbakov

bogdando commented:
Indeed. I believe it is possible to populate master uuid as well

On 09/05/2013 02:39 PM, Mike Scherbakov wrote:

> We don't need UUID of each node. We need only one UUID for master node
> installation.
> 
> On Thu, Sep 5, 2013 at 3:32 PM, Bogdan Dobrelya 
> notifications@github.comwrote:
> 
> > For virtual cases, at least for single KVM host case, it would be, 
> > because
> > it represents the virsh domain name which is always uniq.
> > For HW and other virtual cases, need to be investigated more precisely.
> > 
> > Here is the example of custom fact for uuid would have been 
> > extracted from
> > DMI on bootstrap stage:
> > https://github.com/bogdando/fuelweb/blob/poc_fence/bin/agent#L281-L284
> > 
> > —
> > Reply to this email directly or view it on 
> > GitHubhttps://github.com/Mirantis/fuel/pull/510#issuecomment-23861279
> > .
> 
> ## 
> 
> Mike Scherbakov
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/Mirantis/fuel/pull/510#issuecomment-23861591.

## 

Best regards,
Bogdan Dobrelya,
Researcher TechLead, Mirantis, Inc.
+38 (066) 051 07 53
Skype bogdando_at_yahoo.com
38, Lenina ave.
Kharkov, Ukraine
www.mirantis.com
www.mirantis.ru
bdobrelia@mirantis.com

bogdando commented on deployment/puppet/nailgun/templates/settings.yaml.erb:
```diff
@@ -10,6 +10,8 @@ DATABASE:
   passwd: "<%= database_passwd %>"
 
+FUEL_KEY: "<%= $fuel_key %>"
+
```
typo

e0ne commented:
So should we use 'system-uuid' or 'uuidgen' to generate Fuel key?

Commit:
Fix typo

Commit:
Read Fuel key from /etc/fuel-uuid

mihgen commented:
Lets finally use uuidgen in for example post install script on master,
which is run once, storing this value to some place like /etc/fuel_uuid,
and let's close this discussion.

Mike Scherbakov
On Sep 5, 2013 7:51 PM, "Ivan Kolodyazhny" notifications@github.com wrote:

> So should we use 'system-uuid' or 'uuidgen' to generate Fuel key?
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/Mirantis/fuel/pull/510#issuecomment-23878113
> .

e0ne commented:
retest this please

mirantis-jenkins commented:
:-1: Test FAILed.
Refer to this link for build results: http://fuel-jenkins.mirantis.com/job/fuel_syntax_check/59/

e0ne commented:
retest this please

mirantis-jenkins commented:
:+1: Test PASSed.
Refer to this link for build results: http://fuel-jenkins.mirantis.com/job/fuel_syntax_check/60/

xdel committed:
Merge pull request #526 from sirmax123/heat

added heat module

xdel committed:
Merge pull request #528 from sirmax123/savanna

added savanna module

xdel committed:
Merge pull request #527 from sirmax123/murano

added murano module

xdel committed:
Merge pull request #516 from dmi-try/no-rhel-ks

Remove old versions of cobbler manifests for rhel

angdraug committed:
fixed typos in comments

angdraug committed:
use "cephosd" label for osd to differentiate from osd journal

xdel committed:
Merge pull request #531 from xarses/ceph

Sync ceph working branch

angdraug committed:
parametrized syntax checker

Scan the whole source tree when invoked without parameters, check
specified files when a list of files is passed in parameters.

warpc committed:
Merge pull request #534 from warpc/double_var_declaration

Double var declaration fix for NailyFacts deploy

Commit:
Copy and append key to authorized_keys for users

mihgen commented:
-1
Please resubmit this PR to master

Commit:
Merge pull request #541 from Zipfer/ssh_keys

Copy and append key to authorized_keys for users

xdel committed:
Merge pull request #538 from xarses/syntax-check-parametrized

parametrized syntax checker
